storage/remote/codec_test.go

```diff
@@ -229,3 +232,147 @@ func TestMergeLabels(t *testing.T) {
 		testutil.Equals(t, tc.expected, MergeLabels(tc.primary, tc.secondary))
 	}
 }
+
+func TestChunkSeriesSet(t *testing.T) {
+	b := &bytes.Buffer{}
+	f := &mockedFlusher{}
+	w := NewChunkedWriter(b, f)
+
+	chk1 := tsdbutil.ChunkFromSamples([]tsdbutil.Sample{tsdbutil.NewSample(1, 10), tsdbutil.NewSample(2, 20)})
```

Comment:
NewSample is being used here.

Reply:
But this is wrong as I suggested in the comment or I hope to make. The idea is to rather implement separate `sample{` type here not on tsdbutil.

Reply:
hmm okay. But why is this necessary? :thinking:

Reply:
Hey, sorry I should explain.

We do that for purpose to have clear init statement. Instead of `tsdbutil.NewSample(1, 10)` we can do `sample{ }` which is just much less code to read when we do that a lot. Since we do this everywhere, I feel like we should be make it consistent with rest of the code. IF we want to change it, there should be another PR changing thousands of places like this where we define `type sample struct` etc. BTW that's why tsdbutil.Sample is just interface.

Reply:
still not addressed (:

Reply:
fyi, it has been implemented in one of the previous commits. 
PS: A good practice on my end would be to start commenting 'done' as and when I solve the issues. Haha